zentone/src/main/java/com/github/nisrulz/zentone/wave_generators/WaveByteArrayGenerator.kt

```diff
@@ -1,12 +1,17 @@
 package com.github.nisrulz.zentone.wave_generators
 
+import android.util.Log
 import com.github.nisrulz.zentone.DEFAULT_AMPLITUDE
 import com.github.nisrulz.zentone.DEFAULT_FREQUENCY_HZ
 import com.github.nisrulz.zentone.DEFAULT_SAMPLE_RATE
 import com.github.nisrulz.zentone.internal.minBufferSize
+import kotlin.math.PI
+import kotlin.math.abs
+import kotlin.math.sin
 
-interface WaveByteArrayGenerator {
+abstract class WaveByteArrayGenerator {
```

Comment:
Hmm.. I wouldn't want to do that. The reason this was an interface because I want to have an API surface that is defined by contract and avoid inheritance. This causes issues during testing.